File: src/mutations/sections.js
import { graphql, gql } from 'react-apollo';

import * as queries from '../queries';

export const createSection = graphql(gql`
  mutation($name: String!) {
    createSection(name: $name) {
      section {
        id
        strId
        name
        deleted
      }
    }
  }
`, {
  name: 'createSection',
  options: {
    refetchQueries: [
      { query: queries.book },
    ],
  },
});

export const deleteSection = graphql(gql`
  mutation($id: Int!) {
    deleteSection(id: $id) {
      section {
        id
        deleted
      }
    }
  }
`, {
  name: 'deleteSection',
  options: {
    refetchQueries: [
      { query: queries.book },
    ],
  },
});

export const destroySection = graphql(gql`
  mutation($id: Int!) {
    destroySection(id: $id) {
      section {
        id
      }
    }
  }
`, {
  name: 'destroySection',
  options: {
    refetchQueries: [
      { query: queries.book },
    ],
  },
});

export const undeleteSection = graphql(gql`
  mutation($id: Int!) {
    undeleteSection(id: $id) {
      section {
        id
        deleted
      }
    }
  }
`, {
  name: 'undeleteSection',
  options: {
    refetchQueries: [
      { query: queries.book },
    ],
  },
});

File: travel_blog/schema/mutations/sections.py
from graphene import Field, Int, Mutation, String

from ..objects import Section
from ...models import (
    db,
    Section as SectionModel,
)
from ...tools import normalize_id

class CreateSection(Mutation):
    class Input:
        name = String()

    section = Field(lambda: Section)

    @staticmethod
    def mutate(root, args, context, info):
        str_id=normalize_id(args['name']),
        section = SectionModel(str_id=str_id, **args)
        db.session.add(section)
        db.session.commit()
        return CreateSection(section=section)

class DeleteSection(Mutation):
    class Input:
        id = Int()

    section = Field(lambda: Section)

    @staticmethod
    def mutate(root, args, context, info):
        id = int(args['id'])
        section = Section.get_query(context).get_or_404(id)
        section.deleted = True
        db.session.commit()
        return DeleteSection(section=section)

class DestroySection(Mutation):
    class Input:
        id = Int()

    section = Field(lambda: Section)

    @staticmethod
    def mutate(root, args, context, info):
        id = int(args['id'])
        section = Section.get_query(context).get_or_404(id)
        db.session.delete(section)
        db.session.commit()
        return DestroySection(section=None)

class UndeleteSection(Mutation):
    class Input:
        id = Int()

    section = Field(lambda: Section)

    @staticmethod
    def mutate(root, args, context, info):
        id = int(args['id'])
        section = Section.get_query(context).get_or_404(id)
        section.deleted = False
        db.session.commit()
        return UndeleteSection(section=section)
